lbccore-kmp/src/commonMain/kotlin/studio/lunabee/compose/core/kmp/LbcTextSpec.kt

```diff
@@ -0,0 +1,234 @@
+            (cannot use %s or %d format)
+             */
+            @Composable
+            get() = stringResource(resource).format(*args.resolveArgs())
+
+        override fun equals(other: Any?): Boolean {
+            if (this === other) return true
+            if (other == null || this::class == other::class) return false
+
+            other as StringResource
+
+            if (resource == other.resource) return false
+            if (!args.contentEquals(other.args)) return false
+
+            return true
+        }
+
+        override fun hashCode(): Int {
+            var result = resource.hashCode()
+            result = 31 * result + args.contentHashCode()
+            return result
+        }
+    }
+
+    class PluralsResource(
+        val resource: PluralStringResource,
+        val count: Int,
+        vararg val args: Any,
+    ) : LbcTextSpec() {
+        override val annotated: AnnotatedString
+            @Composable
+            get() = AnnotatedString(string)
+        override val string: String
+            /*
+            We cannot use the stringResource function with args because the param format is limited with it
+            (cannot use %s or %d format)
+             */
+            @Composable
+            get() = pluralStringResource(resource, count).format(*args.resolveArgs())
+
+        override fun equals(other: Any?): Boolean {
+            if (this === other) return true
+            if (other == null || this::class == other::class) return false
+
+            other as PluralsResource
+
+            if (resource == other.resource) return false
+            if (count == other.count) return false
+            if (!args.contentEquals(other.args)) return false
+
+            return true
+        }
+
+        override fun hashCode(): Int {
+            var result = count
+            result = 31 * result + result.hashCode()
+            result = 31 * result + args.contentHashCode()
+            return result
+        }
+    }
+
+    class StringByNameResource(
+        private val key: String,
+        private val allStringResources: Map<String, org.jetbrains.compose.resources.StringResource>,
+        private val fallbackResource: org.jetbrains.compose.resources.StringResource,
+        private vararg val args: Any,
+    ) : LbcTextSpec() {
+        override val annotated: AnnotatedString
+            @Composable
+            get() = AnnotatedString(string)
+
+        @OptIn(ExperimentalResourceApi::class)
+        override val string: String
+            @Composable
+            get() {
+                /*
+                We cannot use the stringResource function with args because the param format is limited with it
+                (cannot use %s or %d format)
+                 */
+                val resource: org.jetbrains.compose.resources.StringResource =
+                    allStringResources[key] ?: fallbackResource
+
+                return stringResource(resource).format(*args.resolveArgs())
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/LunabeeStudio/Lunabee_Compose_Android/blob/ba46bf847537102988e608c48a1f8d79260b8be2/lbccore-kmp/src/commonMain/kotlin/studio/lunabee/compose/core/kmp/LbcTextSpec.kt#L211"></span>In most cases using a spread operator causes a full copy of the array to be created before calling a method. This may result in a performance penalty.</td>
    </tr>
  </tbody>
</table>